src/bin/elementary/embed_config.c

```diff
@@ -0,0 +1,150 @@
+
+#include <assert.h>
+#include <errno.h>
+#include <getopt.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+
+#ifdef HAVE_CONFIG1
+#include "elementary_config.h"
+#endif
+
+static void
+_print_usage(const char *progn, FILE *outf)
+{
+   fprintf(outf, "Usage: %s [options] [input]\n", progn);
+   fprintf(outf, "Options:\n"
+                 "  -o output     specify the filename and path for output\n"
+                 "  -v            show version number\n"
+                 "\n");
+}
+
+char* _read_input_file (FILE* f, const char* fname)
+{
+   fseek(f, 0, SEEK_END);
+   long fs = ftell(f);
+   if (fs < 0)
+     {
+        fprintf(stderr, "eolian: could not get length of '%s'\n", fname);
+        fclose(f);
+        return NULL;
+     }
+   fseek(f, 0, SEEK_SET);
+
+   char *cont = malloc(fs + 1);
+   if (!cont)
+     {
+        fprintf(stderr, "eolian: could not allocate memory for '%s'\n", fname);
+        fclose(f);
+        return NULL;
+     }
+
+   long as = fread(cont, 1, fs, f);
+   if (as != fs)
+     {
+        fprintf(stderr, "eolian: could not read %ld bytes from '%s' (got %ld)\n",
+                fs, fname, as);
+        free(cont);
+        fclose(f);
+        return NULL;
+     }
+
+   cont[fs] = '\0';
+   fclose(f);
+   return cont;
+}
+
+int main(int argc, char** argv)
+{
+   char *output = NULL, *input = NULL;
+   FILE *fin = NULL, *fout = NULL;
+   int err = -1;
+  
+   for (int opt; (opt = getopt(argc, argv, "ho:")) != -1;)
+     switch (opt)
+       {
+        case 0:
+          break;
+        case 'o':
+          output = strdup(optarg);
+          break;
+        case 'h':
+          _print_usage(argv[0], stdout);
```

Comment:
if this will always print on stdout, do we need to set an arg to which file stream to output to?